R/websocket.R

```diff
@@ -108,33 +135,66 @@ null_func <- function(...) { }
 NULL
 
 #' @export
-WebsocketClient <- R6::R6Class("WebsocketClient",
+WebSocket <- R6::R6Class("WebSocket",
   public = list(
+    onMessage = NULL,
+    onOpen = NULL,
+    onClose = NULL,
+    onError = NULL,
     initialize = function(url,
-      onMessage = function(msg) {},
-      onOpen = function() {},
-      onClose = function() {},
-      onFail = function() {},
+      protocols = character(0),
       headers = NULL,
+      autoConnect = TRUE,
       accessLogChannels = c("none"),
-      errorLogChannels = NULL
+      errorLogChannels = c("none")
     ) {
+      self$onOpen <-
+        self$onClose <-
+        self$onError <-
+        self$onMessage <-
+        function(event) {}
```

Comment:
This function will capture the environment of the `initialize` method, so it effectively isn't different from what was there before.

Reply:
Yeah, that's fine. The references to these effectively get cleaned up during removeHandlers (in C++) so it doesn't really matter what they capture.